cmd/src/batch_validate.go

```diff
@@ -36,7 +41,13 @@ Examples:
 			return cmderrors.Usage("additional arguments not allowed")
 		}
 
-		svc := service.New(&service.Opts{})
+		svc := service.New(&service.Opts{
+			Client: cfg.apiClient(apiFlags, flagSet.Output()),
+		})
+
+		if err := svc.DetermineFeatureFlags(ctx); err != nil {
+			return err
+		}
```

Comment:
Just a side-note, no action required: It'd be nice if this is not something that can be forgotten 😬 

```go
package service

func NewWithFeatureFlags(client *api.Client) (Service, error) {
	svc := service.New(&service.Opts{
		Client: client,
	})
	err := svc.DetermineFeatureFlags(ctx)
	return svc, err
}
```

Reply:
Yeah, I was on the fence on this one too. I think I'd rather keep this PR as trivial as it is for now, but we might want to rethink in general how we design and couple the service at some point, since it's not as simple as it once was.